docs/topics/Threads.md

```diff
@@ -0,0 +1,111 @@
+- The [Thread Create](#DOCS_TOPICS_GATEWAY/thread-create) Gateway Event will contain a [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) object as a property of the thread if the current user is a member of, and the user has recently gained access to view the parent channel.
+- The [Thread List Sync](#DOCS_TOPICS_GATEWAY/thread-list-sync) Gateway Event will contain an array of [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) objects for any returned threads the current user is a member of.
+
+### Syncing for other users
+
+> info
+> These require the `GUILD_MEMBERS` [Gateway Intent](#DOCS_TOPICS_GATEWAY/gateway-intents)
+
+- An API `GET` call to `/channels/<channel_id>/thread-members` which returns an array of [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) objects.
+- The [Thread Members Update](#DOCS_TOPICS_GATEWAY/thread-members-update) Gateway Event which will include all users who were added to or removed from a thread by an action.
+
+## Editing & Deleting Threads
+
+Threads can be edited and deleted with the existing `PATCH` and `DELETE` endpoints to edit a channel.
+
+- Deleting a thread requires the `MANAGE_THREADS` permission.
+- Editing a thread to set `archived` to `false` only requires the current user has already been added to the thread. If `locked` is true, then the user must have `MANAGE_THREADS`
+- Editing a thread to change the `name`, `archived`, `auto_archive_duration` fields requires `MANAGE_THREADS` or that the current user is the thread creator
+- Editing a thread to change `rate_limit_per_user` or `locked` requires `MANAGE_THREADS`
+
+## NSFW Threads
+
+Threads do not explicitly set the `nsfw` field. All threads in a channel marked as `nsfw` inherit that setting though.
+
+## New Message Types
+
+Threads introduce a few new [message types](#DOCS_RESOURCES_CHANNEL/message-object-message-types), and re-purpose some others:
+
+- `RECIPIENT_ADD` and `RECIPIENT_REMOVE` have been repurposed to also send when a user is added to or removed from a thread by someone else
+- `CHANNEL_NAME_CHANGE` has been repurposed and is sent when the thread's name is changed
+- `THREAD_CREATED` is a new message sent when a thread is created from an older message, it contains a [message reference](#DOCS_RESOURCES_CHANNEL/message-reference-object-message-reference-structure) with the `guild_id` and `channel_id` of the thread. The `content` of the message is the `name` of the thread.
+- `THREAD_STARTER_MESSAGE` is a new message sent as the first message in certain threads. It _only_ contains a [message reference](#DOCS_RESOURCES_CHANNEL/message-reference-object-message-reference-structure) field that points to the message from which the thread was started.
+
+## Enumerating threads
+
+There are four new `GET` routes for enumerating threads in a specific channel
```

Comment:
I'm not sure why there's need for these 3 endpoints to be seperate:
`/channels/<channel_id>/threads/active`
`/channels/<channel_id>/threads/archived/public`
`/channels/<channel_id>/threads/archived/private`.

It would be more consistent to merge these 3 under a single `/channels/<channel_id>/threads` endpoint and use query parameters such as `state` and `visibility` if needed to be filtered.

However this does add scenarios where someone might try to filter for threads which are active and private. Currently this is not supported for active threads but it should be just as it is for archived ones.

Reply:
I've been polling various folks on this, so far it's actually 100% against having a merged endpoint surprisingly!

I understand the concern, but there's other complexities too, like making rate limits more complicated for example.

Reply:
I don't really mind the endpoints being separate, it's not like it matters much, but I'm confused by this "100% against" poll. One can retrieve all channels within a guild via a single REST request and then filter it client-side, but here we get 3 separate ones just for the sake of filtering?

Why would these have different rate-limits? Doesn't the internal logic differ only by the queried type/archive state?

Reply:
There are absolutely going to be differences.  At a minimum we need pagination here but guild channels doesn't

Reply:
I think the permissions varying are fine, that's the case on e.g. creating messages.

But let's say you went with the single route for threads and had `status` as a field:
If it was set to `archived`, you'd order them by `archive_timestamp` with pagination.
If it was set to `active`, you'd order by `id` without pagination.
And for both active and archived you could just fallback onto ordering by `id` with pagination?

I think this in itself is fine, because the same behavior has to be accounted for when implementing the separate routes. Here we just switch the behavior based on the provided `status` field instead.
However there must be different rate-limits based on the queried thread `status` because, like you said, there will be way more archived threads which makes this merge definitely not worth it.

What about just the `visibility` field though? That'd remove the need for that extra route just to specify `public` or `private` and would let us filter the active threads too.

Reply:
I suspect we won't have different rate limits, I bet all 4 routes end up getting the same rate limit most likely, it's just simpler that way.

Regarding visibility: that's the same thing though as just having two routes, right? idk

Reply:
If they'd have the same rate-limits I'd be all for merging the three endpoints into a single one, but you've said people prefer separate ones.

Regarding visibility: yes, but this also would let us filter the active threads via the field as right now it returns both public and private ones.